client/src/components/SearchInput.js

```diff
@@ -1,16 +1,95 @@
 import React from 'react'
-import { Box, Button, TextInput } from 'grommet'
+import { Box, Button, Select, TextInput, Keyboard } from 'grommet'
+import { useMaterialsSearch } from '../hooks/useMaterialsSearch'
+import { Mappings } from './resources'
+import { sortedObjectKeys } from '../helpers/sortObjectKeys'
+import { getReadable } from '../helpers/readableNames'
+
+export default function SearchInput({
+  initialString = '',
+  onChange,
+  size = 'medium'
+}) {
+  const {
+    query: { search },
+    addFacet,
+    removeFacet,
+    goToSearchResults,
+    setSearchString
+  } = useMaterialsSearch()
+  const [resourceType, setResourceType] = React.useState('ALL')
+  const [string, setString] = React.useState(search || initialString)
+  const resourceOptions = [
+    { label: 'All', value: 'ALL' },
+    ...sortedObjectKeys(Mappings).map((mapping) => {
+      return {
+        label: getReadable(mapping.key),
+        value: mapping.key
+      }
+    })
+  ]
+
+  const submitForm = () => {
```

Comment:
maybe rename to `handleSubmit` or something similar since there is no explicit concept of form here